proposals/lambda-method-group-defaults.md

```diff
@@ -69,41 +81,54 @@ app.MapPost("/todos/{id}", todoHandler);
 ```
 
 ## Current Behavior
-Currently, when a user implements a lambda with a default value, the compiler raises an error stating that default values are not allowed. 
+Currently, when a user implements a lambda with a default or `params` parameter, the compiler raises an error stating that it's not allowed. 
 
 ```csharp
-var addWithDefault = (int addTo = 2) => addTo + 1; // Error: default values not allowed in this context
+var addWithDefault = (int addTo = 2) => addTo + 1; // error CS1065: Default values are not valid in this context.
+var counter = (params int[] xs) => xs.Length; // error CS1670: params is not valid in this context
 ```
 
-When a user attempts to use a method group where the underlying method has a default parameter, the
-default param isn't propagated, so the call to the method doesn't typecheck due to a mismatch in the number of expected parameters.
+When a user attempts to use a method group where the underlying method has a default or `params` parameter, this information isn't propagated, so the call to the method doesn't typecheck due to a mismatch in the number of expected arguments.
 ```cs
-void M(int i = 1) {}
+void M1(int i = 1) { }
+var m1 = M1; // Infers Action<int>
+m1(); // error CS7036: There is no argument given that corresponds to the required parameter 'obj' of 'Action<int>'
 
-var m = M; // Infers Action<int>
-m(); // Error: no value provided for arg0
+void M2(params int[] xs) { }
+var m2 = M2; // Infers Action<int[]>
+m2(); // error CS7036: There is no argument given that corresponds to the required parameter 'obj' of 'Action<int[]>'
 ```
 
-## New Behavior 
+## New Behavior
 
-Following this proposal, default values can be applied to lambda parameters with the following behavior:
+Following this proposal, default values and `params` can be applied to lambda parameters with the following behavior:
 
 ```csharp
 var addWithDefault = (int addTo = 2) => addTo + 1;
 addWithDefault(); // 3
 addWithDefault(5); // 6
+
+var count = (params int[] xs) => xs.Length;
+count(); // 0
+count(1, 2, 3); // 3
 ```
 
-Default values can be applied to method group parameters by specifically defining a method group that
-has a default parameter:
+Default values and `params` can be applied to method group parameters by specifically defining such method group:
 
 ```cs
-void addWithDefault(int addTo = 2) {
+int addWithDefault(int addTo = 2) {
   return addTo + 1;
 }
 
 var add1 = addWithDefault; 
-add1(); // ok, default parameter will be used.
+add1(); // ok, default parameter will be used
+
+int count(params int[] xs) {
```

Comment:
While we're here, can you do a pass to make all of these method names follow C# `PascalCase` convention?

Reply:
Good idea. 322230de0c13a49e980d67dc0ebe5e54ed26a04d